docs/developer/available-plugins/adyen.mdx

```diff
@@ -0,0 +1,269 @@
+ log in to their bank's website or redirect to payment provider page, Saleor will set `confirmationNeeded` to `true` 
+ and provide all Adyen's action data required to finalize payment as a `confirmationData`.
+
+```json
+{
+  "data": {
+    "checkoutComplete": {
+      "order": {
+        "id": "T3JkZXI6MjkxMw=="
+      },
+      "confirmationNeeded": false,
+      "confirmationData": "{\"paymentData\": \"Ab02b4c0!BQABAgA1Xk1nQmoiOJCEtVVds....4n8OzNsVJGN3XHg7kMg=\",\"paymentMethodType\": \"scheme\",\"url\": \"https:\\/\\/checkoutshopper-test.adyen.com\\/checkoutshopper\\/threeDS2.shtml\",\"data\": {\"MD\": \"M2RzM...DRi\",\"PaReq\": \"BQABAg....Rz3hacE\",\"TermUrl\": \"http:\\/\\/mirumee.com\\/plugins\\/mirumee.payments.adyen\\/additional-actions?payment=UGF5bWVudDoxMDM%3D&checkout=08546530-4718-422c-a912-9a1642a9031c\"},\"method\": \"POST\",\"type\": \"redirect\"}"
+    }
+  }
+}
+```
+
+After processing additional action by the customer, call `checkoutComplete` a second time and pass data received from Adyen (`state.data`) as a `paymentData` field:
+
+```graphql
+mutation {
+  checkoutComplete(
+    token: "14f11c1c-9bc7-4145-89f7-67b60ef943f2"
+    paymentData: "{\"additional-action\": \"data recieved from onAdditionalDetails\" ...,\"}"
+  ) {
+    order {
+      id
+    }
+    confirmationNeeded
+    confirmationData
+  }
+}
+```
+The response of the mutation will contain a newly created order or additional data required for the next additional action from a customer. 
+In case, when more actions are required, process them in the same way as the second `checkoutComplete` call.
+
+#### Processing a redirect additional action
+
+if `action.type` is `redirect`, Adyen will redirect a customer to a page where the customer can finalize the payment.
+ After that customer will be redirected back. The request will go through the Saleor, which will confirm the current status of payment. 
+ As a next step, Saleor will use `returnUrl` provided in `checkoutPaymentCreate` mutation to redirect shopper to store page.
+The `returnUrl` will have attached below parameters:
+  - Required:
+    - `checkout` - the ID of processed checkout.
+    - `payment` - the ID of processed payment.
+    - `resultCode` - status of the payment recieved from Adyen. Can be used to determine the current status of the payment. [Docs](https://docs.adyen.com/online-payments/payment-result-codes)
+  - Optional:
+    - `action` -  action data if an additional action is required by customer.
+
+If payment was successfully processed, Saleor will create an order. The next mutation `checkoutComplete` call will return `order` in the response.
+
+
+
+## Enabling native 3DS2
+Saleor by default uses redirect 3D Secure 1 and 2. To use native 3DS, [enable it in the Saleor's dashboard](dashboard/configuration/plugins/adyen.md#enabling-the-adyen-integration). 
+Setting this flag will include to Adyen's request `additionalData.allow3DS2: true`, to all card transactions.
```

Comment:
@korycins This sentence is pretty unclear to me.

Reply:
On the dashboard side, there is a checkbox `Enable native 3D Secure`  that determines if native 3DS is enabled or not. 
Setting this checkbox to True will include flag `additionalData.allow3DS2: true` to all "payment processing" requests that we send to Adyen.